pkl-gradle/src/main/java/org/pkl/gradle/task/BasePklTask.java

```diff
@@ -146,6 +148,14 @@ public Provider<String> getEvalRootDirPath() {
   @Optional
   public abstract MapProperty<URI, URI> getHttpRewrites();
 
+  @Input
+  @Optional
+  public abstract MapProperty<String, ExternalReader> getExternalModuleReaders();
+
+  @Input
+  @Optional
+  public abstract MapProperty<String, ExternalReader> getExternalResourceReaders();
```

Comment:
Be aware that `@Optional`, according to [its documentation](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/Optional.html), is intended to be used for file-based properties. It either will do nothing or will fail plugin validation if it is used with non-file properties (the exact behavior here might depend on Gradle version which is used to build the plugin, I think). I would recommend just dropping it - empty map is always a valid value (and is actually the default value) for map properties anyway.